File: LsHcore/src/main/pl/edu/osp/objects/Plant.java
package pl.edu.osp.objects;


public abstract class Plant implements StaticObject {
    protected byte bio = 0;
    protected short hp = 0;
    final PlantConstans plantConst;
    
    public Plant(PlantConstans pc) {
    	plantConst = pc;
    }
    
    public void nextDay() {
        yields();
    }
       
    public byte getBio() { return bio; }
    public void setBio(byte b) { bio = b; }
    public short getHP() {return hp;}
	public byte maxBio() {return plantConst.mBio;}
	public byte growLev() {return plantConst.gLev;}
	public short maxHP() {	return plantConst.mHP;}
	public String name() {return plantConst.name;}
	@Override
	public String toString() {
	    return plantConst.name + " " + hp + " " + bio;
	}
    
    public void yields() {
        bio += plantConst.gLev;
        if(bio > plantConst.mBio)
            bio = plantConst.mBio;     
    }
    
    
    
}

File: LsHcoreScala/src/main/scala/eu/brosbit/immovable/Plant.scala
package eu.brosbit.immovable

abstract  class Plant extends ImmovableObject {
  val growBio:Short
  val maxBio:Short
  val growHP: Short
  val maxHP:Short
  val name:String


  protected var bio: Short = 0
  protected var hp: Short = 0

  def getBio = bio
  def setBio(bio:Short) = this.bio = bio
  def setHP(hp:Short) = this.hp = hp
  def getHP() = hp

  override def nextDay() = yields()


  def yields() = {
    bio = (bio + growBio).toShort
    hp = (hp + growHP).toShort
    if(bio > maxBio) bio = maxBio
    if(hp > maxHP) hp = maxHP
  }


}
